### ISSUE TITLE
allow to pass file name & file ending to consume method

### ISSUE THREAD

**Issue by kremalicious** [oceanprotocol/squid-js#177]
At the moment, squid-js returns a streamed data blob which will end up as `undefinedfile-0` on the user's file system at the end of the `ocean.assets.consume()` method.

### Expected behavior

As a library consumer, I can pass an optional file name, e.g. `almond-sales-data.csv` like so:

```js
const file = await ocean.assets.consume(
    agreementId,
    ddo.id,
    service.serviceDefinitionId,
    accounts[0],
    '',
    'almond-sales-data.csv',
    index
)
```

This file name is then used for the binary blob pushed to the user.

**Comment by ssallam** [comment]
We can improve the default behaviour by keeping the file names in the ddo `files` dict. When removing the original urls at time of publishing the asset, we can add the file names so they can be used to name the downloaded file.

**Comment by kremalicious** [comment]
Sounds great and I think having both would be great too, default to original file name from publishing, but still allow passing a custom name

**Comment by Pedro-vk** [comment]
Taking into account the changes of #184 I think that is not going to be needed because is going to use the name of the asset. What do you think?

**Comment by kremalicious** [comment]
Use case to get rid of the default names without file ending so that's definitely nicely solved with #184 and all the other component changes. While passing a custom name would still be nice we could save that for later, when that use case is actually coming up. So feel free to close this issue then until this happens

**Comment by kremalicious** [comment]
not needed right now
